src/Appium.Net/Appium/AppiumOptions.cs

```diff
@@ -78,75 +79,111 @@ public new string BrowserName
         {
             Dictionary<string, object> knownOptions = new Dictionary<string, object>();
 
-            if (!string.IsNullOrEmpty(this.App))
+            if (!string.IsNullOrEmpty(App))
             {
-                knownOptions[AppOption] = this.App;
+                knownOptions[AppOption] = App;
             }
 
-            if (!string.IsNullOrEmpty(this.AutomationName))
+            if (!string.IsNullOrEmpty(AutomationName))
             {
-                knownOptions[AutomationNameOption] = this.AutomationName;
+                knownOptions[AutomationNameOption] = AutomationName;
             }
 
-            if (!string.IsNullOrEmpty(this.DeviceName))
+            if (!string.IsNullOrEmpty(DeviceName))
             {
-                knownOptions[DeviceNameOption] = this.DeviceName;
+                knownOptions[DeviceNameOption] = DeviceName;
             }
 
-            if (!string.IsNullOrEmpty(this.PlatformVersion))
+            if (!string.IsNullOrEmpty(PlatformVersion))
             {
-                knownOptions[PlatformVersionOption] = this.PlatformVersion;
+                knownOptions[PlatformVersionOption] = PlatformVersion;
             }
 
             return knownOptions;
-
         }
 
         private Dictionary<string, object> BuildAppiumOptionsDictionary()
         {
             var appiumOptions = BuildAppiumKnownOptionsDictionary();
 
-            foreach (KeyValuePair<string, object> pair in this.additionalAppiumOptions)
+            foreach (KeyValuePair<string, object> pair in additionalAppiumOptions)
             {
                 appiumOptions.Add(pair.Key, pair.Value);
             }
 
             return appiumOptions;
         }
 
+        /// <summary>
+        /// Converts the current instance of <see cref="AppiumOptions"/> to a ReadOnlyDictionary.
+        /// </summary>
+        /// <returns>A ReadOnlyDictionary representation of the <see cref="AppiumOptions"/>.</returns>
         public IDictionary<string, object> ToDictionary()
         {
-            var writeable = this.GenerateDesiredCapabilities(true);
-            return (writeable.AsReadOnly() as ReadOnlyDesiredCapabilities).ToDictionary();
+            IWritableCapabilities writeable = GenerateDesiredCapabilities(true);
+            var baseDict = (writeable.AsReadOnly() as ReadOnlyDesiredCapabilities).ToDictionary();
+            return MergeOptionsDictionary(baseDict);
         }
+
+        /// <summary>
+        /// Merges the provided <see cref="IWritableCapabilities"/> dictionary with the <see cref="AppiumOptions"/> dictionary.
+        /// </summary>
+        /// <param name="baseDict">The base <see cref="IWritableCapabilities"/> dictionary.</param>
+        /// <returns>A ReadOnlyDictionary representing the merged <see cref="AppiumOptions"/> dictionary.</returns>
+        private IDictionary<string, object> MergeOptionsDictionary(IDictionary<string, object> baseDict)
```

Comment:
this could be done via LINQ: https://stackoverflow.com/questions/294138/merging-dictionaries-in-c-sharp

Reply:
done.